File: app/src/main/java/com/example/solom/rememberizer/CustomButton.java
package com.example.solom.rememberizer;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.widget.ImageButton;

public class CustomButton extends ImageButton {

    private int content;

    private boolean shown;

    private BitmapFactory.Options bmpOptions;

    public CustomButton(Context context) {
        super(context);
        bmpOptions = new BitmapFactory.Options();
        bmpOptions.inSampleSize = 5;
        setScaleType(ScaleType.FIT_XY);
        setImageBitmap(BitmapFactory.decodeResource(getContext().getResources(), R.drawable.backgr, bmpOptions));
    }

    public boolean isShown(){
        return shown;
    }

    public int getContent() {
        return content;
    }

    public void setContent(int content) {
        this.content = content;
    }

    public void setShown() {
        shown = !shown;
        if(shown){
            setImageBitmap(BitmapFactory.decodeResource(getContext().getResources(), content, bmpOptions));
        }
        else {
            setImageBitmap(BitmapFactory.decodeResource(getContext().getResources(), R.drawable.backgr, bmpOptions));
        }
    }
}

File: app/src/main/java/com/example/solom/rememberizer/CustomButton.kt
package com.example.solom.rememberizer

import android.content.Context
import android.content.res.Resources
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.widget.ImageButton
import android.widget.ImageView

class CustomButton(context: Context) : ImageButton(context) {

    var content: Int = 0
    private var shown: Boolean = false
    private val bmpOptions: BitmapFactory.Options

    init {
        bmpOptions = BitmapFactory.Options()
        bmpOptions.inSampleSize = 5
        scaleType = ImageView.ScaleType.FIT_XY
        setImageBitmap(BitmapFactory.decodeResource(getContext().resources, R.drawable.backgr, bmpOptions))
    }

    override fun isShown(): Boolean {
        return shown
    }

    fun setShown() {
        shown = !shown
        if (shown) {
            setImageBitmap(BitmapFactory.decodeResource(context.resources, content, bmpOptions))
        } else {
            setImageBitmap(BitmapFactory.decodeResource(context.resources, R.drawable.backgr, bmpOptions))
        }
    }
}
